test/dpul_collections/indexing_pipeline_test.exs

```diff
@@ -30,6 +30,38 @@ defmodule DpulCollections.IndexingPipelineTest do
         IndexingPipeline.get_hydration_cache_entry!(hydration_cache_entry.id)
       end
     end
+
+    test "write_hydration_cache_entry/1 upserts a cache entry" do
+      {:ok, record} =
+        IndexingPipeline.write_hydration_cache_entry(%{
+          data: %{},
+          source_cache_order: ~U[2024-07-23 20:05:00Z],
+          cache_version: 0,
+          record_id: "some record_id"
+        })
+
+      {:ok, record_2} =
+        IndexingPipeline.write_hydration_cache_entry(%{
+          data: %{},
+          source_cache_order: ~U[2024-07-24 20:05:00Z],
+          cache_version: 0,
+          record_id: "some record_id"
+        })
+
+      {:ok, record_3} =
+        IndexingPipeline.write_hydration_cache_entry(%{
+          data: %{},
+          source_cache_order: ~U[2024-07-22 20:05:00Z],
+          cache_version: 0,
+          record_id: "some record_id"
+        })
+
+      reloaded = IndexingPipeline.get_hydration_cache_entry!(record_2.id)
+      assert record.cache_order != reloaded.cache_order
+      record_3 = IndexingPipeline.get_hydration_cache_entry!(record_2.id)
```

Comment:
Something doesn't look right in this test. we're fetching record_3 from record_2's id here? Also they all have the same `record_id` value -- I don't think that's relevant but I think it's a little confusing.